activerecord/lib/active_record/associations.rb

```diff
@@ -229,7 +229,16 @@ def association(name) #:nodoc:
       association = association_instance_get(name)
 
       if association.nil?
-        unless reflection = self.class._reflect_on_association(name)
+        reflection = self.class._reflect_on_association(name)
```

Comment:
At a glance, this sounds like too high-level of a place to do this: aren't other places expecting `_reflect_on_association` to have a complete view of the available associations?